gnd/src/main/java/com/google/android/gnd/workers/TileDownloadWorker.java

```diff
@@ -42,106 +45,106 @@
     super(context, params);
     this.context = context;
     this.localDataStore = localDataStore;
-    this.tileId = params.getInputData().getString(TILE_ID);
-  }
-
-  /** Creates input data for the TileDownloadWorker. */
-  public static Data createInputData(String tilePrimaryKey) {
-    return new Data.Builder().putString(TILE_ID, tilePrimaryKey).build();
   }
 
   /**
    * Given a tile, downloads the given {@param tile}'s source file and saves it to the device's app
    * storage. Optional HTTP request header {@param requestProperties} may be provided.
    */
-  private Result downloadTileFile(Tile tile, Optional<HashMap<String, String>> requestProperties) {
-    try {
-      URL url = new URL(tile.getUrl());
-      HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-
-      if (requestProperties.isPresent()) {
-        for (Map.Entry<String, String> property : requestProperties.get().entrySet()) {
-          connection.setRequestProperty(property.getKey(), property.getValue());
-        }
-      }
-
-      connection.connect();
-
-      InputStream is = connection.getInputStream();
-      FileOutputStream fos = context.openFileOutput(tile.getPath(), Context.MODE_PRIVATE);
-      byte[] byteChunk = new byte[BUFFER_SIZE];
-      int n;
-
-      while ((n = is.read(byteChunk)) > 0) {
-        fos.write(byteChunk, 0, n);
+  private void downloadTileFile(Tile tile, Map<String, String> requestProperties)
+      throws IOException {
+    URL url = new URL(tile.getUrl());
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+    if (!requestProperties.isEmpty()) {
+      for (Map.Entry<String, String> property : requestProperties.entrySet()) {
+        connection.setRequestProperty(property.getKey(), property.getValue());
       }
+    }
 
-      is.close();
-      fos.close();
-
-      localDataStore
-          .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.DOWNLOADED).build())
-          .blockingAwait();
-
-      return Result.success();
-
-    } catch (IOException e) {
-      Log.d(TAG, "Failed to download and write file.", e);
+    connection.connect();
 
-      localDataStore
-          .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.FAILED).build())
-          .blockingAwait();
+    InputStream is = connection.getInputStream();
+    FileOutputStream fos = context.openFileOutput(tile.getPath(), Context.MODE_PRIVATE);
+    byte[] byteChunk = new byte[BUFFER_SIZE];
+    int n;
 
-      return Result.failure();
+    while ((n = is.read(byteChunk)) > 0) {
+      fos.write(byteChunk, 0, n);
     }
-  }
-
-  /** Update a tile's state in the database and initiate a download of the tile source file. */
-  private Result downloadTile(Tile tile) {
-    localDataStore
-        .insertOrUpdateTile(tile.toBuilder().setState(Tile.State.IN_PROGRESS).build())
-        .blockingAwait();
 
-    return downloadTileFile(tile, Optional.empty());
+    is.close();
+    fos.close();
```

Comment:
Does this need to be in a `finally` block?

Reply:
Used a try-with-resources block instead (since android studio suggested it).

Reply:
Done see 2b81c0b